ucm2/sof-soundwire/HiFi.conf

```diff
@@ -10,25 +10,31 @@ SectionVerb {
 
 If.spkdev {
 	Condition {
-		Type String
-		Empty "${var:SpeakerCodec1}"
+		Type RegexMatch
+		Regex "(rt722(-sdca)?)"
+		String "${var:SpeakerCodec1}"
 	}
+	True.Include.spkdev.File "/sof-soundwire/${var:SpeakerCodec1}-spk.conf"
 	False.Include.spkdev.File "/sof-soundwire/${var:SpeakerCodec1}.conf"
 }
 
 If.micdev {
 	Condition {
-		Type String
-		Empty "${var:MicCodec1}"
+		Type RegexMatch
+		Regex "(rt722(-sdca)?)"
+		String "${var:MicCodec1}"
 	}
+	True.Include.micdev.File "/sof-soundwire/${var:MicCodec1}-mic.conf"
 	False.Include.micdev.File "/sof-soundwire/${var:MicCodec1}.conf"
 }
 
 If.hsdev {
 	Condition {
-		Type String
-		Empty "${var:HeadsetCodec1}"
+		Type RegexMatch
+		Regex "(rt722(-sdca)?)"
+		String "${var:HeadsetCodec1}"
 	}
+	True.Include.hsdev.File "/sof-soundwire/${var:HeadsetCodec1}-jack.conf"
```

Comment:
@perexg this needs your review.

RT722 is one of the first 'integrated' codecs with multiple "functions" in the same chip. Before we had one function per chip, but some for the same rt722 we have mic,jack,amp capabilities.

So we probably need a mechanism to identify such functions without changing the main filters.

Reply:
I think we don't need to split rt722.conf into rt711-spk.conf, rt722-mic.conf, and rt722-sdca-jack.conf. We can put SectionDevice."Mic" and SectionDevice."Speaker" in rt722.conf and SectionDevice."Headphones" and SectionDevice."Headset" in rt722-sdca.conf.

Reply:
Or we could add suffix in machine driver, and use rt722-amp.conf, rt722-jack.conf, and rt722-mic.conf.